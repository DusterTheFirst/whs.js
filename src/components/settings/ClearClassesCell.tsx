/*!
 * Copyright (C) 2018-2020  Zachary Kohnen (DusterTheFirst)
 */

import { useNavigation } from "@react-navigation/core";
import { StackNavigationProp } from "@react-navigation/stack";
import React, { useContext } from "react";
import { LegacyTempClassesContext } from "../../contexts";
import { SettingsParams } from "../../navigators/SettingsNavigator";
import { RedCell } from "../../styles/components/tableview";
import { clearClassesAlert } from "../../util/alerts";

/** A settings cell to clear the classes */
export default function ClearClassesCell() {
    const navigation = useNavigation<StackNavigationProp<SettingsParams>>();
    const tempClasses = useContext(LegacyTempClassesContext);

    const clear = () => clearClassesAlert(() => {
        tempClasses.clear();

        navigation.navigate("ClassesList");
    });

    return (
        <RedCell title="Clear Classes" onPress={clear} />
    );
}