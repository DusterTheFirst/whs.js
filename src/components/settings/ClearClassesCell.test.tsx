/*!
 * Copyright (C) 2018-2019  Zachary Kohnen (DusterTheFirst)
 */

import React from "react";
import { render } from "react-native-testing-library";
import ClearClassesCell from "./ClearClassesCell";

// TODO: Test press

it("Renders <ClearClassesCell/>", () => {
    const comp = render(<ClearClassesCell/>);

    expect(comp.toJSON()).toMatchSnapshot();
});