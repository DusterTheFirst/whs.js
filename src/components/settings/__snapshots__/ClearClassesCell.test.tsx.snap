// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <ClearClassesCell/> 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#FFF",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "minHeight": 44,
            "paddingLeft": 15,
            "paddingRight": 20,
          },
          Object {
            "backgroundColor": "#FFF",
          },
          Object {},
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingVertical": 10,
          }
        }
      >
        <Text
          allowFontScaling={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "flex": 1,
                "fontSize": 16,
                "letterSpacing": -0.32,
              },
              Object {
                "color": "#000",
              },
              Object {
                "color": "#FF5050",
              },
            ]
          }
        >
          Clear Classes
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
